# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: toolkit/components/antitracking/ContentBlocking.cpp
# Commit: a537b565801d
# Full Hash: a537b565801dec42739cc6be68b50c4a5517775e
# Author: Dimi Lee <dlee@mozilla.com>
# Date: 2020-04-25 09:49:23
# Description:
#   Bug 1632815 - Check IsThirdPartyPrincipal only when we have an in-process browsing context r=timhuang
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D72377
# ==============================================================================

diff -r 2eff17c17783 -r a537b565801d toolkit/components/antitracking/ContentBlocking.cpp
--- a/toolkit/components/antitracking/ContentBlocking.cpp	Wed Apr 22 04:18:01 2020 +0000
+++ b/toolkit/components/antitracking/ContentBlocking.cpp	Fri Apr 24 13:16:27 2020 +0000
@@ -318,12 +318,16 @@
     // Only continue to run in child processes when aParentContext is
     // in-process and tracking origin is not third-party with respect to
     // the parent window.
-    bool isThirdParty;
-    nsCOMPtr<nsIPrincipal> principal =
-        AntiTrackingUtils::GetPrincipal(aParentContext);
-    Unused << trackingPrincipal->IsThirdPartyPrincipal(principal,
-                                                       &isThirdParty);
-    runInSameProcess = aParentContext->IsInProcess() && !isThirdParty;
+    if (aParentContext->IsInProcess()) {
+      bool isThirdParty;
+      nsCOMPtr<nsIPrincipal> principal =
+          AntiTrackingUtils::GetPrincipal(aParentContext);
+      Unused << trackingPrincipal->IsThirdPartyPrincipal(principal,
+                                                         &isThirdParty);
+      runInSameProcess = !isThirdParty;
+    } else {
+      runInSameProcess = false;
+    }
   }
 
   if (runInSameProcess) {
